Add Stepper component

joshuatf opened the pull request:
Fixes #2164 

Adds in a Muriel style stepper component to be used with the onboarding profiler.

### Thoughts
* Made steps that are prior to the `currentStep` complete by default unless a boolean is passed specifying otherwise.  Still allows the `isComplete` boolean for a step to override this and/or mark the final step complete.
* I did my best to match this design around other Muriel components, but this is not an existing component with guidelines so it's a bit mix and match.  Obviously it's contrasted quite heavily with existing wc-admin components, but I believe that's the direction we'll be headed in for most components going forward.

### Screenshots
<img width="424" alt="Screen Shot 2019-05-16 at 5 09 23 PM" src="https://user-images.githubusercontent.com/10561050/57841653-8860fe80-77fd-11e9-85d6-aab23a5f9719.png">
<img width="632" alt="Screen Shot 2019-05-16 at 5 09 12 PM" src="https://user-images.githubusercontent.com/10561050/57841655-8860fe80-77fd-11e9-8450-6fd6d808c23d.png">

### Detailed test instructions:

1.  Open the devdocs `wp-admin/admin.php?page=wc-admin#/devdocs`.
2.  Go through the steps in the stepper component.
3.  Make sure it works as expected.
4.  Make sure styling is correct.

justinshreve commented:
@joshuatf One more note -- I think we need to update the changelog for the components package as well.

timmyc commented on packages/components/src/stepper/example.md:
```diff
@@ -0,0 +1,68 @@
+        steps.forEach( s => s.isComplete = true );
+    }
+
+	return (
```
minor tabbing a bit out of sorts here with the above.

joshuatf replied:
Thanks, Timmy!  Updated 😄

timmyc commented:
This is looking great - thanks for... stepping up... to build this new component Josh.

</dadjokes>

psealock commented on packages/components/src/stepper/style.scss:
```diff
@@ -0,0 +1,71 @@
+
+		&.is-active,
+		&.is-complete {
+			color: #6802f3;
```
The designs look like they introduce a new color scheme. Now (or in follow up) seems like a good time to introduce some variables.

joshuatf replied:
Great idea @psealock- added in https://github.com/woocommerce/woocommerce-admin/pull/2246/commits/1ea32e1013836d0a342c390400710b26ef7332f4

joshuatf commented:
Thanks for the ✅ @justinshreve!  Added font sizing and updated colors to the new Figma designs.  I imagine we might need change the naming of the color vars in here (`$muriel-active`) once we start adding in more components, but I think this will do for now.
